csrc/multidevice/collective.h

```diff
@@ -0,0 +1,172 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+#ifdef USE_DISTRIBUTED
+
+#include <multidevice/communicator.h>
+#include <multidevice/multidevice.h>
+
+namespace nvfuser {
+
+/*
+  This struct gathers all the parameters necessary for the
+  construction a collective
+*/
+struct TORCH_CUDA_CU_API CollectiveParams {
+  DeviceIdxType root = -1;
+  std::vector<at::Tensor> src_bufs;
+  std::vector<at::Tensor> dst_bufs;
+  std::vector<DeviceIdxType> team; // should not have duplicate
```

Comment:
Here also shouldn't this be a vector of `RankType`s?

Reply:
No, the team is inded a set of devices that will participate to the collectives.

In general, now RankType shouldn't be used outside of the Communicator: as far as nvFuser is concerned, there is no ranks but just device indices.